File: Task2/apz-pzpi-21-11-milennyi-ivan-task2/src/view_models/feed_supply_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct FeedSupplyVM {
    id: u64,
    amount: u64,
    timestamp: u64,
    storekeeper_name: Option<String>,
    storekeeper_surname: Option<String>,
}

File: Task4/apz-pzpi-21-11-milennyi-ivan-task4/src/viewModels/FeedSupplyVM.ts
import { timestampToDate } from "../utils/helpers";

export class FeedSupplyVM {
    id: number;
    amount: number;
    date: string;
    storekeeperName: string | null;
    storekeeperSurname: string | null;

    constructor(id: number, amount: number, timestamp: number, storekeeperName: string | null, storekeeperSurname: string | null) {
        this.id = id;
        this.amount = amount / 1000;
        this.storekeeperName = storekeeperName;
        this.storekeeperSurname = storekeeperSurname;
        this.date = timestampToDate(timestamp);
    }
}
